modules/core/src/main/scala/org/tesselation/infrastructure/trust/TrustDaemon.scala

```diff
@@ -0,0 +1,74 @@
+package org.tesselation.infrastructure.trust
+
+import cats.Parallel
+import cats.effect.std.Random
+import cats.effect.{Async, Spawn, Temporal}
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tesselation.config.types.TrustDaemonConfig
+import org.tesselation.domain.Daemon
+import org.tesselation.domain.trust.storage.TrustStorage
+import org.tesselation.kryo.KryoSerializer
+import org.tesselation.schema.peer.PeerId
+import org.tesselation.security.SecurityProvider
+
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+trait TrustDaemon[F[_]] extends Daemon[F]
+
+object TrustDaemon {
+
+  def make[F[_]: Async: SecurityProvider: KryoSerializer: Random: Parallel](
+    cfg: TrustDaemonConfig,
+    trustStorage: TrustStorage[F],
+    selfPeerId: PeerId
+  ): TrustDaemon[F] = new TrustDaemon[F] {
+
+    private val logger = Slf4jLogger.getLogger[F]
+
+    def start: F[Unit] =
+      for {
+        _ <- Spawn[F].start(modelUpdate.foreverM).void
+      } yield ()
+
+    private def calculateTrust(): F[Unit] = {
+      val predicted = trustStorage.getTrust.map { trust =>
+        val fixed = trust.toList
+        val selfTrust = fixed.flatMap { case (peerId, trustInfo) => trustInfo.publicTrust.map(t => peerId -> t) }
+        val allNodes = fixed.map {
+          case (peerId, trustInfo) =>
+            peerId -> trustInfo.peerLabels
+        } :+ (selfPeerId -> selfTrust.toMap)
+        val peerIdToIdx = allNodes.zipWithIndex.map { case ((peerId, _), idx) => peerId -> idx }.toMap
+        val idxToPeerId = peerIdToIdx.map { case (k, v) => v -> k }
+        val idMappedScores: Map[PeerId, Double] = SelfAvoidingWalk
```

Comment:
`val idMappedScored` is redundant because you immediately return it in line 59

Reply:
Fixed